tests/test_encoding/test_decision_tree_encoder.py

```diff
@@ -2,24 +2,97 @@
 import pandas as pd
 import pytest
 
+from sklearn.exceptions import NotFittedError
+
 from feature_engine.encoding import DecisionTreeEncoder
 
 
-def test_encoding_method_param(df_enc):
-    # defaults
+# init parameters
+@pytest.mark.parametrize("enc_method", ["count", False, 1])
+def test_error_if_encoding_method_not_permitted_value(enc_method):
+    msg = (
+        "`encoding_method` takes only values 'ordered' and 'arbitrary'."
+        f" Got {enc_method} instead."
+    )
+    with pytest.raises(ValueError) as record:
```

Comment:
pytest.raises(ValueError, match=msg) is way cleaner

Reply:
thanks!